pkgs/by-name/cc/ccusage/package.nix

```
@@ -0,0 +1,63 @@
+{
+  stdenv,
+  lib,
+  fetchFromGitHub,
+  bun,
+}:
+
+let
+  ccusage-deps = stdenv.mkDerivation rec {
+    pname = "ccusage-deps";
+    version = "13.0.1";
+    src = fetchFromGitHub {
+      owner = "ryoppippi";
+      repo = "ccusage";
+      rev = "v${version}";
+      hash = "sha256-5bqcY33h9sycWHY553WDQwFADFaPPg+Ph+sPCQPuPoI=";
+    };
+    nativeBuildInputs = [ bun ];
+    buildPhase = ''
+      bun install --no-save --ignore-scripts
+    '';
+    installPhase = ''
+      mkdir -p $out
+      cp -r node_modules $out/
+    '';
+    outputHash = "sha256-TzKNKDxXzYP8p82Upms7wUqGZUGggPLX7IuUfQprXGA=";
+    outputHashAlgo = "sha256";
+    outputHashMode = "recursive";
+  };
+
+in
+stdenv.mkDerivation rec {
+  pname = "ccusage";
+  version = "13.0.1";
+  src = fetchFromGitHub {
+    owner = "ryoppippi";
+    repo = "ccusage";
+    rev = "v${version}";
+    hash = "sha256-5bqcY33h9sycWHY553WDQwFADFaPPg+Ph+sPCQPuPoI=";
+  };
+  nativeBuildInputs = [ bun ];
+  buildPhase = ''
+    ln -s ${ccusage-deps}/node_modules ./node_modules
+  '';
+  installPhase = ''
+          mkdir -p $out/bin $out/share/ccusage
+          cp -r . $out/share/ccusage/
+          cat > $out/bin/ccusage << EOF
+    #!/bin/bash
+    cd $out/share/ccusage
+    exec ${bun}/bin/bun run ./src/index.ts "\$@"
+    EOF
+          chmod +x $out/bin/ccusage
+  '';
```

```suggestion
  buildPhase = ''
    runHook preBuild

    ln -s ${ccusage-deps}/node_modules ./node_modules

    runHook postBuild
  '';
  installPhase = ''
    runHook preInstall

    mkdir -p $out/bin $out/share/ccusage
    cp -r . $out/share/ccusage/
    cat > $out/bin/ccusage << EOF
    #!/bin/bash
    cd $out/share/ccusage
    exec ${bun}/bin/bun run ./src/index.ts "\$@"
    EOF
    chmod +x $out/bin/ccusage

    runHook postInstall
  '';
```
Don't revoke user's ability to extend phases without overwriting them.